src/program.cpp

```diff
@@ -229,6 +254,10 @@ bool LineIsWhitespace(std::string& line)
     return true;
 }
 
+bool LineIsComment(std::string &line)
```

Comment:
I believe you need to check if the first non-whitespace/tab symbol is an octothorpe `#`. It could be the case that someone does
```
# comment
    # weirdly indented comment
```

Reply:
fixed